### PR TITLE
WindowCovering: Cleanup enum xml

### PR DESCRIPTION
**Pull request by jmeg-sfy**
#### Problem
What is being fixed?  Examples:
* Cleanup Enum declaration that were in the wrong namespace using Ember
* Fix Bug with LCD mutex lock at Init 

#### Change overview
This PR cleanup the code base from #define constant value to use kConstant enum
Also we update the LCD code to use uint16_t instead of uint8_t
DISABLE some minor test on WNCV 2.1 due to ZAP giving issues comparing enum with CheckConstraintNotValue
invalid operands to binary expression ('chip::app::Clusters::WindowCovering::EndProductType' and 'int')
CheckConstraintNotValue("endProductType", endProductType, 250)

#### Testing
How was this tested? (at least one bullet point required)
* Tested on EFR32 demo
* Unit tests passes minus the 2 minor steps disabled that need to be address

### CODE REVIEW

**Review by jmartinez-silabs — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

File: src/app/tests/suites/certification/Test_TC_WNCV_2_1.yaml [hunk 1]
```diff
@@ -50,6 +50,7 @@ tests:
           error: UNSUPPORTED_WRITE
 
     - label: "3b: reads back the RO mandatory attribute: Type"
+      disabled: true
```
Comment (bzbarsky-apple) [inline src/app/tests/suites/certification/Test_TC_WNCV_2_1.yaml:53]: Why are tests being disabled?

File: src/app/zap-templates/zcl/data-model/chip/window-covering.xml [hunk 2]
```diff
@@ -126,11 +126,12 @@ limitations under the License.
   </cluster>
 
   <!-- Window Covering Descriptive Enum Set -->
-  <enum name="WcType" type="ENUM8">
```
Comment (bzbarsky-apple) [inline src/app/zap-templates/zcl/data-model/chip/window-covering.xml:129]: Please remove `WcType` from the list in `src/app/zap-templates/templates/app/helper.js`.  Same for `WcEndProductType`.

File: src/app/tests/suites/certification/Test_TC_WNCV_2_1.yaml (hunk 1 above)
Comment (jmeg-sfy) [reply src/app/tests/suites/certification/Test_TC_WNCV_2_1.yaml:53]: https://github.com/project-chip/connectedhomeip/issues/16995

Comment (jmeg-sfy) [reply src/app/tests/suites/certification/Test_TC_WNCV_2_1.yaml:53]: There is a problem with the constraint check generate template @bzbarsky-apple

File: src/app/zap-templates/zcl/data-model/chip/window-covering.xml (hunk 2 above)
Comment (jmeg-sfy) [reply src/app/zap-templates/zcl/data-model/chip/window-covering.xml:129]: wasn't aware of this list in helper.js ; thanks i have just addressed this !

**Review by woody-apple — APPROVED** [review]

**Pull request merged**